src/applications/facility-locator/components/search-results-items/VaFacilityResult.jsx

```diff
@@ -31,7 +42,7 @@ const VaFacilityResult = ({
           markerText={location.markerText}
         />
         {/* eslint-disable-next-line jsx-a11y/interactive-supports-focus, jsx-a11y/no-static-element-interactions */}
-        <span onClick={clickHandler} onKeyUp={clickHandler}>
+        <span onClick={clickHandler} onKeyUp={keyboardHandler}>
```

Comment:
In prod, when you used your keyboard to tab to a result, the event for clicking a result was fired even though it was just being focused, not selected. Added a specific keyboard handler for the search result name to look for only the Enter key and the space bar key.

Screenshots below of prod behavior. I tabbed to "North Central Federal VA Clinic" and the `fl-results-click` event fired
<img width="1040" alt="Screenshot 2024-08-08 at 12 10 11 PM" src="https://github.com/user-attachments/assets/eb508c7c-0ca3-4823-b821-ed04b1d28111">

<img width="547" alt="Screenshot 2024-08-08 at 12 10 00 PM" src="https://github.com/user-attachments/assets/ebe1e8ab-af1a-49e8-bc32-63f799f5ed02">